api/envoy/config/trace/v3/zipkin.proto

```diff
@@ -68,4 +68,11 @@ message ZipkinConfig {
   // Optional hostname to use when sending spans to the collector_cluster. Useful for collectors
   // that require a specific hostname. Defaults to :ref:`collector_cluster <envoy_v3_api_field_config.trace.v3.ZipkinConfig.collector_cluster>` above.
   string collector_hostname = 6;
+
+  // If this is set to true, then Envoy will be treated as independent service. Server span will be created
```

Comment:
Can you clarify what "independent" means in this context?

Reply:
> Can you clarify what "independent" means in this context?

In the previous implementation of zipkin, sidecar envoy will be treated as part of the App. Following trace chain will be generated.

```
server span (inbound sidecar) -> APP (May generates span or not) -> client span (outbound sidecar).
```

And if the `start_child_span` is set to true, the following trace chain will be generated, **and which is wrong**.

```
server span (inbound sidecar) -> server span (router of inbound sidecar) -> APP -> client span (outbound sidecar) -> client span (router of outbound sidecar).

or

server span (gateway) -> server span (router of gateway) -> APP
```

And in our new implementation, if the `independent_proxy` is set to true, **the envoy (inbound or outbound) will be treated as independent hop** in the trace chain. Server span will be created for downstream request and client span will be created for upstream request.

If the `start_child_span` is set to true, the following trace chain will be generated.


```
server span (inbound sidecar) -> client span (router of inbound sidecar) -> APP -> server span (outbound sidecar) -> client span (router of outbound sidecar).

or

server span (gateway) -> client span (router of gateway) -> APP

```

Reply:
It is still somewhat not clear to me and could be due to my misunderstanding of zipkin and tracing.
Is this field separating server and client spans?